models/content_test.go

```diff
@@ -0,0 +1,125 @@
+package models_test
+
+import (
+	"testing"
+
+	"github.com/ONSdigital/dp-topic-api/apierrors"
+	"github.com/ONSdigital/dp-topic-api/models"
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestContentValidation(t *testing.T) {
+	Convey("Given an empty content, it is successfully validated", t, func() {
+		content := models.Content{
+			State: models.StateCreated.String(), //!!! remove topic out of the enum name
+		}
+		err := content.Validate()
+		So(err, ShouldBeNil)
+	})
+
+	Convey("Given an content with no state supplied, it fails to validate  with the expected error", t, func() {
+		content := models.Content{}
+		err := content.Validate()
+		So(err, ShouldResemble, apierrors.ErrTopicInvalidState)
+
+	})
+
+	Convey("Given an content with a state that does not correspond to any expected state, it fails to validate with the expected error", t, func() {
+		content := models.Content{
+			State: "wrong",
+		}
+		err := content.Validate()
+		So(err, ShouldResemble, apierrors.ErrTopicInvalidState)
+	})
+
+	Convey("Given a fully populated valid content with a valid download variant, it is successfully validated", t, func() {
```

Comment:
I dont understand what 'valid download variant' means in this context? should it actually be 'valid state'?

Reply:
This has been removed in next PR, after this is done.